src/main/java/com/redhat/devtools/lsp4ij/hint/LSPNavigationLinkHandler.java

```diff
@@ -35,27 +36,41 @@
  */
 public class LSPNavigationLinkHandler extends TooltipLinkHandler  {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(LSPNavigationLinkHandler.class);//$NON-NLS-1$
+    public static final String PREFIX_OR_SUFFIX = "#lsp-navigation/";
 
-    private static final String PREFIX = "#lsp-navigation/";
-    public static final String POS_SEPARATOR = ";";
+    public static final String HASH_SEPARATOR = "#";
 
     @Override
-    public boolean handleLink(@NotNull String refSuffix, @NotNull Editor editor) {
-        int pos = refSuffix.lastIndexOf(':');
-        if (pos <= 0 || pos == refSuffix.length() - 1) {
-            LOGGER.info("Malformed suffix: " + refSuffix);
-            return true;
-        }
+    public boolean handleLink(@NotNull String fileUrl,
+                              @NotNull Editor editor) {
+        return handleLink(fileUrl, null, editor.getProject());
+    }
 
-        String uri = refSuffix.substring(0, pos);
-        Range range = toRange(refSuffix.substring(pos + 1));
-        Location location = new Location();
-        location.setUri(uri);
-        if (range != null) {
-            location.setRange(range);
-        }
-        return LSPIJUtils.openInEditor(location, editor.getProject());
+    /**
+     * Open in an editor the given fileUrl and tries to create the file if it doesn't exist.
+     *
+     * <p>
+     *     the fileUrl can have following syntax:

```

Comment:
the following syntax is supported for fileUrl

Reply:
fixed